### ISSUE TITLE
Non-interactive mode

### ISSUE THREAD

**Issue by kawasaki-kanagawa** [leotaku/kojirou#38]
This is more of a question: is any interest to support non-interactive mode? I'm setting up kojirou to be run on-demand via CI pipeline and found some parts which I wish work differently (and I'm happy to contribute the changes as well as Woodpecker CI plugin if desired). Cheers!

# Progress bar output

![Progress bar output on non-interactive tty is effectively gibberish.](https://github.com/leotaku/kojirou/assets/150632067/454b7963-dd7b-49e6-b9ac-3631dc6f6d91)

1. Would be nice if the progress bar output would switch to something simpler when the program detects it's running in non-interactive mode.

1. On the same note but to the opposite direction, would be fantastic if program can output color text when it detects CI environment variable — modern CI systems support color, but not "delete" characters. Something along the lines of:
    ```go
    if os.Getenv("CI") != "" {
      color.NoColor = false;
    }
    ```

# External store

Because of on-demand download in ephemeral runner, I would want to have the books to be stored in external storage such as S3. 

Right now, my pipeline goes from kojirou to [Rclone](https://rclone.org), which is fine except kojirou does not take advantage of whether some volumes can be skipped. While I can hack on it to be wrapped further, I think this would be a pretty neat feature to be built in the program.

**Comment by leotaku** [comment]
> Would be nice if the progress bar output would switch to something simpler when the program detects it's running in non-interactive mode.

Yeah, that does sound like a good idea. The full progress bar should probably only be drawn if the output is a TTY.

> On the same note but to the opposite direction, would be fantastic if program can output color text when it detects CI environment variable — modern CI systems support color, but not "delete" characters. Something along the lines of:

Probably a no on that one. Seems like a bad idea to assume that running in CI means that colors are supported. Maybe we could support a new option or something like [CLICOLORS](http://bixense.com/clicolors/), but even that seems like it's probably not worth the additional code, seeing how colors in a CI environment seems like a very niche use-case.

> External store

Probably a no for me as well, as this would almost certainly require additional dependencies and code. Maybe you could mount your external storage as a FUSE filesystem to achieve the desired effect?
